core/runtime/src/test/java/io/quarkus/logging/GenerateLog.java

```diff
@@ -24,11 +24,21 @@ public class GenerateLog {
             " * Invocations of all {@code static} methods of this class are, during build time, replaced by invocations\n" +
             " * of the same methods on a generated instance of {@link Logger}.\n" +
             " */\n" +
-            "public final class Log";
+            "public final class Log {\n" +
+            "    // automatically generated by io.quarkus.logging.GenerateLog";
+    // the condition here is an attempt to stop IntelliJ flagging the Log methods as always throwing
     private static final String LOG_METHOD_BODY = ") {\n" +
-            "        throw fail();\n" +
+            "        if (always()) {\n" +
+            "            throw fail();\n" +
+            "        } else {\n" +
+            "            throw fail();\n" +
+            "        }\n" +
             "    }";
     private static final String FAIL_METHOD = "" +
+            "\n" +
+            "    private static boolean always() {\n" +
+            "        return true;\n" +
+            "    }\n" +
```

Comment:
IntelliJ can't follow this? If so, better for us :)

Reply:
I wouldn't be surprised if it was because they need the control flow analysis to finish very quickly.